source/dshow-dialogbox.cpp

```diff
@@ -0,0 +1,50 @@
+#include "dshow-dialogbox.hpp"
+#include "log.hpp"
+
+namespace DShow {
+    DeviceDialogBox::DeviceDialogBox() :
+        deviceFilter(nullptr),
+        threadId(0),
+        threadHandle(nullptr),
+        isOpen(false)
+    {
+    }
+
+    void DeviceDialogBox::Open(IUnknown* filter) {
+        if (!isOpen) {
+            deviceFilter = filter;
+            threadHandle = CreateThread(nullptr, 0, CallCreate, (void*) this, 0, &threadId);
+
+            if (threadHandle == nullptr) {
+                Warning(L"Could not create thread for filter dialog box");
+            }
+        }
+    }
+
+    void DeviceDialogBox::Close() {
+        if (isOpen) {
+            PostThreadMessage(threadId, WM_QUIT, 0, 0);
+            CloseHandle(threadHandle);
+            isOpen = false;
+        }
+    }
+
+    DWORD DeviceDialogBox::Create() {
+        ComQIPtr<ISpecifyPropertyPages> pages(deviceFilter);
+        CAUUID cauuid;
+
+        if (pages != nullptr) {
+            if (SUCCEEDED(pages->GetPages(&cauuid)) && cauuid.cElems) {
+                isOpen = true;
+                OleCreatePropertyFrame(nullptr, 0, 0, nullptr, 1,
+                                       (LPUNKNOWN *)&deviceFilter,
+                                       cauuid.cElems, cauuid.pElems, 0,
+                                       0, nullptr);
+                CoTaskMemFree(cauuid.pElems);
+            }
+        }
+
+        isOpen = false;
```

Comment:
Wondering why we are assigning isOpen to false here, are we not going through this code if we correctly created the window?

Reply:
It is because the call to OleCreatePropertyFrame blocks the thread and the thread only continues if the dialog box is closed manually by the user. I set it isOpen to false here to prevent an unnecessary call to PostThreadMessage on the Close function if the dialog box is closed manually

Reply:
I see, since we are also doing it in the `Close()` function, is one of these 2 calls are redundant?

Reply:
Removed the check in the `Create` function. Keep in mind that there is one way to close a dialog box window that is not the regular win-dshow configuration window. For example the new C270 configuration window does not block the whole application and also do not prevent SLOBS from closing, but it stays open anyway. The only way I can think to handle these cases is by searching all the processes open in the system for the right one to close, but without knowing the name of the process it is next to impossible